Update round navigation player, styles

pjsier opened the pull request:
## Description

Significant updates to the round navigation player incorporating feedback up to this point

## Screenshots



https://github.com/user-attachments/assets/ff6d446d-d0f4-4be9-b27f-c762adeaeddf



![round-select-update-mobile](https://github.com/user-attachments/assets/dbff312b-bb8b-4c96-b40f-bc068d0e1dfa)
![round-select-update](https://github.com/user-attachments/assets/cfe9a3b5-ccbe-43ea-be08-28a5e53cf415)

pjsier commented:
@artoonie no rush on this, but looks like the Heroku CI build is failing and the logs don't show up for me. Happy to modify any failing tests to use the new setup

Tests in on `testLiveBrowserHeadless` are passing for me locally, but given the amount of markup that was changed I wouldn't be surprised if this has downstream impacts on other tests. I'm seeing some failures in `testLiveBrowserWithHead` that seem like they could be related, but not 100% sure

artoonie commented:
Hi Pat, I've migrated Heroku to a Teams setup and provided access -- this is my first time using Heroku Teams, so please let me know if you've received an email to join the team and can view the tests.

The only test failure I see is this one, which is currently failing on `main` too, so I'll look into this myself to figure out what's going on:
```
FAIL: test_election_pages (electionpage.tests.ElectionPageTests.test_election_pages)
The election pages are error-free for non-scrapable election pages.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/electionpage/tests.py", line 122, in test_election_pages
    self._ensure_eventually_asserts(lambda: self.assertEqual(bargraphIframeWrapper.find_element(
  File "/app/visualizer/tests/liveServerTestBaseClass.py", line 180, in _ensure_eventually_asserts
    assertion()
  File "/app/electionpage/tests.py", line 122, in <lambda>
    self._ensure_eventually_asserts(lambda: self.assertEqual(bargraphIframeWrapper.find_element(
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: '403px' != '328px'
- 403px
+ 328px
```

Some additional context:
Throughout RCVis you'll see a handful of oddly-specific tests like this one. The goal here is to catch any unintentional changes, not to mandate this level of precision.

artoonie commented:
The above issue has been resolved with https://github.com/artoonie/rcvis/pull/539 -- if you pull in main, the tests should pass.

I think this is very close! I've created a Review App and here are three visualizations on it:
https://rcvis-pr-538.herokuapp.com/v/multi_winner2
https://rcvis-pr-538.herokuapp.com/v/portland-2015-mayoral-race
https://rcvis-pr-538.herokuapp.com/v/2013-minneapolis-mayor

Some edge cases I've found:
1. If the description text is too short, it can shift to the right of the vis:
![image](https://github.com/user-attachments/assets/9d7b6a81-f397-4be8-86c5-fb0884a5d04c)

2. Is it easy to try to get an even number of columns per row? I find that this would look better if it were distributed evenly:
![image](https://github.com/user-attachments/assets/f47ec65b-6200-4796-8aa1-ba3cc93cc5ae)

3. Thanks for the update to stop the text from changing during animation! A final tweak to keep the existing functionality unchanged: previously, once the animation would start, the text would change to "After __ rounds of this __ Ranked Choice Voting election, ___ were elected. Read the FAQs or play the animation" and remain static on that text. Now, it just remains on the text it had before the animation began playing. Can you change it to the initial summary text during animation? Let me know if you need more details.

4. No action on this yet, but flagging early: there was a lot of talk on whether we want the button to say "Play"/"Stop" or something else. We'll likely request an update but I don't think we have a consensus yet.

Once these are done, can we pull in Temi to schedule a Bug Bash?

Thanks again for the work here -- this is a huge visual improvement to RCVis and I'm excited to deploy it!

pjsier commented:
Thanks for the review! Rebased and made some updates including cleaning up references to the old bar chart slider and using the new `playing()` method here

Responding to your specific points here

1. Good catch! I added `flex-column` as a class here. I thought that was included in Bootstrap, but it didn't show up until I manually added it, but I'm happy to remove that if it's unneeded
2. I think this is going to be tricky. From looking into flex and grid layouts it doesn't seem like there's a way out of the box to do this, so there would need to be custom JS for the layout and I have a feeling that could introduce a lot more unexpected behavior than it resolves. Open to thinking this through more! Just seems like it could be difficult
3. Sounds great! Should be updated now
4. No worries on that at all! Given the amount of content/layout changes here I'm expecting there to be some back and forth on the specifics so that's fine with me

artoonie commented:
Okay, PR #539 did fix one issue, but I see that the test is still failing, and that makes sense: the new round selector has a different height than the old one, so I expect the height of the iframe to be different than it was before. I suspect all that's left is to update 335 to 403px on `electionpage/tests.py:123`.

No problem on the grid, let's defer that.

Code looks solid and I see no additional visual issues, so let me coordinate with Temi on a final review here.

pjsier commented:
Made some changes based on feedback! This still doesn't look quite right yet, but wanted to push for now since a review app will be easier to try out. "Previous Round" and "Next Round" look a bit wordy in context, and they're also not the same text width, so I wonder if "Play rounds" and then having them just be arrows makes sense?

![2025-04-12_08-01](https://github.com/user-attachments/assets/c1a5c54c-d3cb-4f43-bf81-4f73c0a7d911)

artoonie commented:
I agree with your gut -- here's an idea:
![image](https://github.com/user-attachments/assets/a4056913-699b-41c4-86d4-d6296c5e82fd)

1. space-between with 20px left and right padding on the prev/next/play
2. include round numbers instead of previous and next
3. 50% scale on the SVG
4. "Auto-Step" as text (as ever, likely to change)

Thoughts on this? cc @slhudson

artoonie commented:
This looks awesome and I have no further notes. Code looks good. Will approve once I get the thumbs up from a couple stakeholders. 🚀

pjsier commented:
Sounds great! I can address any feedback once that comes in. I think switching "Auto-Step" back to "Play" might help with space on mobile because currently that's wrapping, but open to any ideas there once others have a chance to take a look

slhudson commented:
Hi folks,

I think we need to take a step back on the button labels. Now that I’ve had
a chance to play with it in in a working model — esp on mobile — it’s
become clear to me that it’s the most prominent feature on the page, so
whatever words go there affect how the user orients to the entire screen.

It would be helpful to me if that could be a conversation and not a thread.

Sally

On Thu, Apr 17, 2025 at 7:49 AM Pat Sier ***@***.***> wrote:

> Sounds great! I can address any feedback once that comes in. I think
> switching "Auto-Step" back to "Play" might help with space on mobile
> because currently that's wrapping, but open to any ideas there once others
> have a chance to take a look
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/artoonie/rcvis/pull/538#issuecomment-2812620773>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEB6N3I6WEOS4OIMUJYN6RT2Z6IK3AVCNFSM6AAAAAB2PWUGG6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJSGYZDANZXGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *pjsier* left a comment (artoonie/rcvis#538)
> <https://github.com/artoonie/rcvis/pull/538#issuecomment-2812620773>
>
> Sounds great! I can address any feedback once that comes in. I think
> switching "Auto-Step" back to "Play" might help with space on mobile
> because currently that's wrapping, but open to any ideas there once others
> have a chance to take a look
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/artoonie/rcvis/pull/538#issuecomment-2812620773>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEB6N3I6WEOS4OIMUJYN6RT2Z6IK3AVCNFSM6AAAAAB2PWUGG6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJSGYZDANZXGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

artoonie reviewed (approved):
💯  thanks Pat! I know a couple small changes are pending, but code is ready and the tests are all in place

pjsier commented:
@artoonie sounds great, I just pushed that change and this is ready to merge whenever works!
